src/get-glamor-classname.js

```diff
@@ -25,31 +25,32 @@ function extractGlamorStyles(className = '') {
 
 export default getGlamorClassName
 
-function getGlamorClassName(styles, props, cssOverrides, theme, context) {
+function getGlamorClassName({
+  styles,
+  props,
+  cssOverrides,
+  cssProp,
+  theme,
+  context,
+}) {
+  const {
+    glamorStyles: parentGlamorStyles,
+    glamorlessClassName,
+  } = extractGlamorStyles(props.className)
   const {mappedArgs, nonGlamorClassNames} = handleStyles(
-    styles,
+    [...styles, parentGlamorStyles, cssOverrides, cssProp],
```

Comment:
Yes, and this preserves the original presidence. We have tests for it which you can check out. I'd link you to them, but I'm on mobile.